Added tile system and test level.

Nikolar97 opened the pull request:
Added a tile system and a test level for it.

Nikolar97 committed:
Adding the basic tile system

Nikolar97 committed:
Merge branch 'team/fried_cucumber/main' into team/fried_cucumber/feature/tile_system

Nikolar97 committed:
Added the first level for testing and tweaked the z axis for the sprite draw priority.

Nikolar97 committed:
Fixed camera and world scale

UbiBelETF commented on source/dagger/gameplay/team_game/hero_character.cpp:
```diff
@@ -0,0 +1,6 @@
+#include "hero_character.h"
+
+hero_character::hero_character()
```
This...
a) isn't needed because the class doesn't exist
b) doesn't compile because of the class not existing

UbiBelETF commented on source/dagger/gameplay/team_game/tilemap_legends.cpp:
```diff
@@ -0,0 +1,121 @@
+	sprite.position = { x_ * x_step, y_ * y_step, 90 };
+	AssignSprite(sprite, "spritesheets:tiles_dungeon:crate");
+	return entity;
+}
```
There are many problems here but I want team members to review. Do not merge this as-is.

Nikolar97 committed:
Removed hero_character, visual studio just delisted it

Nikolar97 committed:
Adding other features to tile system.
Merge branch 'team/fried_cucumber/main' into team/fried_cucumber/feature/tile_system

Nikolar97 committed:
Merged and added collision
